src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/PayRateServiceTest.kt

```diff
@@ -455,12 +456,61 @@ class PayRateServiceTest {
       val request = listOf(PayRateRequest(incentiveLevel = "STD", payBand = "A", rate = BigDecimal(3.2)))
 
       assertThatThrownBy {
-        payRatesService.buildNewPayRates(request, courseActivity)
+        payRatesService.buildNewPayRates(request, courseActivity, null)
       }
         .isInstanceOf(BadDataException::class.java)
         .hasMessageContaining("Pay band code A does not exist")
     }
 
+    @Test
+    fun `Changing the end date of the activity`() {
+      // The default existing activity has 1 active pay rate - iepLevel = "STD", payBand = "5", halfDayRate = 3.2
+      val request = listOf(
+        PayRateRequest("STD", "2", BigDecimal(0.3)),
+        PayRateRequest("STD", "5", BigDecimal(3.2)),
+      )
+
+      val newPayRates = payRatesService.buildNewPayRates(
+        request,
+        courseActivity.copy(scheduleEndDate = threeDaysAhead),
+        tomorrow,
+      )
+
+      assertThat(newPayRates.size).isEqualTo(2)
```

Comment:
nicer to combine all three `assertThat` into single one extracting `endDate` and comparing to `threeDaysAhead`